command/job/move.go

```diff
@@ -0,0 +1,403 @@
+package job
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+	"sync"
+
+	"github.com/hashicorp/nomad/api"
+	"github.com/hashicorp/nomad/scheduler"
+	"github.com/mitchellh/colorstring"
+	log "github.com/sirupsen/logrus"
+	"github.com/urfave/cli"
+)
+
+func Move(c *cli.Context, logger *log.Logger) error {
+	jobName := c.Args().First()
+
+	// Sanity check
+	if jobName == "" {
+		return fmt.Errorf("must provide a job name or prefix")
+	}
+	if c.String("constraint") == "" {
+		return fmt.Errorf("must provide new constrain name")
+	}
+	if c.String("operand") == "" {
+		return fmt.Errorf("must provide new constrain name")
+	}
+	if c.String("value") == "" {
+		return fmt.Errorf("must provide new constrain name")
+	}
+
+	newConstraint := api.NewConstraint(fmt.Sprintf("${%s}", c.String("constraint")), c.String("operand"), c.String("value"))
+
+	// create Nomad API client
+	nomadClient, err := api.NewClient(api.DefaultConfig())
+	if err != nil {
+		return err
+	}
+
+	jobsToMove := []string{jobName}
+
+	// if we stop by prefix, then query for all the jobs
+	if c.Bool("as-prefix") {
+		jobsToMove = []string{}
+
+		jobs, _, err := nomadClient.Jobs().List(&api.QueryOptions{Prefix: jobName})
+		if err != nil {
+			return err
+		}
+
+		for _, job := range jobs {
+			if excludeFilter := c.String("exclude"); excludeFilter != "" {
```

Comment:
rather than nesting another if statement, just check for exclude filter and if matches, continue